frame/contracts/src/exec.rs

```diff
@@ -75,6 +75,45 @@ where
 	}
 }
 
+/// combined key type for both fixed and variable storage keys.
+pub enum Key<T: Config> {
+	/// variant for fixed sized keys.
+	Fix(FixSizedKey),
+	/// variant for variable sized keys.
+	Var(VarSizedKey<T>),
+}
+
+impl<T: Config> Key<T> {
+	/// copies self into a new vec.
+	pub fn to_vec(&self) -> Vec<u8> {
+		match self {
+			Key::Fix(v) => v.to_vec(),
+			Key::Var(v) => v.to_vec(),
+		}
+	}
+}
+
+impl<T: Config> From<FixSizedKey> for Key<T> {
+	fn from(v: FixSizedKey) -> Self {
+		Key::Fix(v)
+	}
+}
+
+impl<T: Config> From<VarSizedKey<T>> for Key<T> {
+	fn from(v: VarSizedKey<T>) -> Self {
+		Key::Var(v)
+	}
+}
+
+impl<T: Config> StorageKey<T> for Key<T> {
+	fn hash(&self) -> Vec<u8> {
+		match self {
+			Key::Fix(v) => <[u8; 32] as StorageKey<T>>::hash(v),
```

Comment:
Why not use the alias here?

```suggestion
			Key::Fix(v) => <FixedSizedKey as StorageKey<T>>::hash(v),
```